torch/_inductor/codegen/common.py

```diff
@@ -36,6 +44,114 @@ def index_prevent_reordering(index: typing.List[sympy.Expr], index_vars, sizes):
     return [*index, sympy_dot(index_vars, FlexibleLayout.contiguous_strides(sizes))]
 
 
+def _data_type_propagation(sub_graph: torch.fx.Graph):
+    def propagate_node(node: torch.fx.Node):
+        _node: torch.fx.Node = node
+        if _node.target == "ops":
+            return False
+        ops_to_bool = [
+            "is_inf",
+            "is_nan",
+            "bitwise_xor",
+            "logical_not",
+            "signbit",
+            "le",
+            "lt",
+            "ge",
+            "gt",
+            "eq",
+            "ne",
+            "bitwise_not",
+            "bitwise_or",
```

Comment:
I suppose the result can be either bool or integer? Why it has to be bool?
https://pytorch.org/docs/stable/generated/torch.bitwise_or.html

Reply:
Thanks for reminding here, I removed `bitwise_not, bitwise_or, bitwise_left_shift, bitwise_right_shift` here and also add UT for this.